File: python/bob/bob.py
import re
def response(hey_bob):
    is_q = len(hey_bob) > 0 and re.search("\?\s*$", hey_bob)
    silence = re.search("^\s*$", hey_bob)
    yelling = not silence and not re.search('[a-z]', hey_bob) and re.search('[A-Z]', hey_bob)
    if is_q and yelling:
        return 'Calm down, I know what I\'m doing!'
    elif is_q:
        return 'Sure.'
    elif yelling:
        return 'Whoa, chill out!'
    elif silence:
        return 'Fine. Be that way!'
    else:
        return 'Whatever.'

File: ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    is_question = remark.match?(/\?\s*\z/)
    is_silence = remark.match?(/\A\s*\z/)
    is_yelling = remark.gsub(/[\W\d\s]/,'').match?(/^[A-Z]+$/)

    return "Calm down, I know what I'm doing!" if is_yelling and is_question
    return "Whoa, chill out!" if is_yelling
    return "Sure." if is_question
    return "Fine. Be that way!" if is_silence
    return "Whatever."
  end
end
